src/Hadolint/Rules.hs

```diff
@@ -1164,3 +1166,21 @@ noIllegalInstructionInOnbuild = instructionRule code severity message check
     check (OnBuild (From _)) = False
     check (OnBuild (Maintainer _)) = False
     check _ = True
+
+relativeCopyWithoutWorkdir :: Rule
+relativeCopyWithoutWorkdir = instructionRuleState code severity message check ("", [])
```

Comment:
I think the state would be more cleanly represented, and likely more performant, using a `Map Text (Set Text)` it will allow you to lookup by image alias and check in a single operation that a dir already exists in the Set

Reply:
I made it even simpler by using a `Map Text Bool`, since this rule really only cares about whether or not a `WORKDIR` is set and not what that is set to.